src/main/java/de/tum/in/www1/artemis/web/rest/hestia/ExerciseHintResource.java

```diff
@@ -85,12 +84,12 @@ public ExerciseHintResource(ExerciseHintRepository exerciseHintRepository, Autho
 
         // Reload the exercise from the database as we can't trust data from the client
         Exercise exercise = exerciseRepository.findByIdElseThrow(exerciseHint.getExercise().getId());
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, exercise, null);
```

Comment:
You should check for the `exerciseId` here and not the one linked to the exerciseHint. Also, this should be done as the first check as it's the first part of the route.